core/src/server/rpc_module.rs

```diff
@@ -853,9 +853,10 @@ impl SubscriptionSink {
 	}
 
 	/// Attempt to accept the subscription and respond the subscription method call.
+	/// The subscription ID is returned on success.
 	///
 	/// Fails if the connection was closed, or if called multiple times.
-	pub fn accept(&mut self) -> Result<(), SubscriptionAcceptRejectError> {
+	pub fn accept(&mut self) -> Result<RpcSubscriptionId<'static>, SubscriptionAcceptRejectError> {
```

Comment:
what's the benefit of returning `SubscriptionId` here instead of having `fn subscription_id(&self) -> Option<RpcSubscritionId<'static>>` as well?

It's rather edge-case to access the subscription id and I think 90% of the use cases doesn't need to know this...

Reply:
Yep indeed, this should be an edge-case and not the most common one.
It does make sense to have a dedicated method for this.
Thanks for having a look! 👍